app/Helpers/database/ticket.php

```diff
@@ -302,40 +320,33 @@ function countRequestTicketsByUser(?User $user = null): int
 
 function countOpenTicketsByDev(User $dev): array
 {
-    $cacheKey = CacheKey::buildUserOpenTicketsCacheKey($dev->User);
-
-    return Cache::remember($cacheKey, Carbon::now()->addHours(20), function () use ($dev) {
```

Comment:
This query is so fast (~3ms worst case I observed) it doesn't seem like there's any point in caching it.

Reply:
It definitely mattered six months ago (see #1928). It looks like the only thing in the [eloquent](https://github.com/RetroAchievements/RAWeb/pull/1928/files#diff-33c2242e14e47d19d23e3afcebf5459b73b9ed4d1f0f07c0ce57605d9df67d77L434-L443) that changed is the filter on `achievement.Author` -> `achievement->user_id`. Maybe there was an index added?

I do still see the query being duplicated, but there's only one query, and it takes ~0.5ms.

Reply:
> Maybe there was an index added?

Digging a little deeper into this, `Author` had an index too. However, I believe the perf discrepancy comes down to the fact that `Author` wasn't a foreign key, whereas `user_id` is. When the DB engine knows about a foreign key relationship, the query optimizer can more efficiently execute the join and oftentimes avoid a full table scan.